src/EnergyPlus/HVACDXSystem.cc

```diff
@@ -2398,22 +2384,15 @@ namespace HVACDXSystem {
                         // If humidity setpoint is not satisfied and humidity control type is CoolReheat,
                         // then overcool to meet moisture load
 
-                        if ((OutletHumRatDXCoil > DesOutHumRat) && (PartLoadFrac < 1.0) &&
-                            (DXCoolingSystem(DXSystemNum).DehumidControlType == DehumidControl_CoolReheat)) {
+                        if ((OutletHumRatDXCoil > DesOutHumRat) && (DXCoolingSystem(DXSystemNum).DehumidControlType == DehumidControl_CoolReheat)) {
 
-                            //           IF NoLoadHumRatOut is lower than (more dehumidification than required) or very near the DesOutHumRat,
-                            //           do not run the compressor
-                            if ((NoLoadHumRatOut - DesOutHumRat) < HumRatAcc) {
-                                // PartLoadFrac = PartLoadFrac; // keep part-load fraction from sensible calculation // Self-assignment commented out
-                                //           If the FullLoadHumRatOut is greater than (insufficient dehumidification) or very near the DesOutHumRat,
-                                //           run the compressor at PartLoadFrac = 1.
-                            } else if ((DesOutHumRat - FullLoadHumRatOut) < HumRatAcc) {
+                            if ((NumOfSpeeds > 1 && SpeedNum == NumOfSpeeds && SpeedRatio == 1.0) || (NumOfSpeeds == 1 && CycRatio == 1.0)) {
```

Comment:
Then similar approach to solving as with temperature noting what speed was needed during temperature control and whether or not more speeds are available for additional dehumidification capacity.